memory/src/stark.rs

```diff
@@ -1,20 +1,25 @@
 use crate::columns::MemoryCols;
+use crate::{MachineWithMemBus, MemoryChip};
 use core::borrow::Borrow;
-use p3_air::{Air, AirBuilder, PermutationAirBuilder};
-use p3_field::AbstractField;
-use p3_matrix::Matrix;
+use valida_machine::{chip, ValidaAirBuilder};
 
-#[derive(Default)]
-pub struct MemoryStark {}
+use p3_air::{Air, AirBuilder, PermutationAirBuilder};
+use p3_field::PrimeField;
+use p3_matrix::MatrixRows;
 
-impl<AB: PermutationAirBuilder> Air<AB> for MemoryStark {
+impl<F, M, AB> Air<AB> for MemoryChip
+where
+    F: PrimeField,
+    M: MachineWithMemBus<F = F>,
+    AB: ValidaAirBuilder<F = F, Machine = M>,
+{
     fn eval(&self, builder: &mut AB) {
         self.eval_main(builder);
     }
 }
 
-impl MemoryStark {
-    fn eval_main<AB: PermutationAirBuilder>(&self, builder: &mut AB) {
+impl MemoryChip {
+    fn eval_main<F: PrimeField, AB: PermutationAirBuilder<F = F>>(&self, builder: &mut AB) {
```

Comment:
Seems like `F` isn't needed since we can refer to `AB::F`, though `F` is shorter, did you want to shorten the uses below?

Reply:
Ah, no, will remove. I think I had copied that line from another stark that was using a method in `PrimeField`.